ddtrace/bootstrap/sitecustomize.py

```diff
@@ -142,6 +52,25 @@ def cleanup_loaded_modules(aggressive=False):
 
 log = get_logger(__name__)
 
+if os.environ.get("DD_GEVENT_PATCH_ALL") is not None:
+    deprecate(
+        "The environment variable DD_GEVENT_PATCH_ALL is deprecated and will be removed in a future version. ",
+        postfix="There is no special configuration necessary to make ddtrace work with gevent if using ddtrace-run. "
+        "If not using ddtrace-run, import ddtrace.auto before calling gevent.monkey.patch_all().",
+        removal_version="2.0.0",
+    )
+if "gevent" in sys.modules or "gevent.monkey" in sys.modules:
+    import gevent.monkey  # noqa
+
+    if gevent.monkey.is_module_patched("threading"):
+        warnings.warn(
+            "Loading ddtrace after gevent.monkey.patch_all() is not supported and is "
+            "likely to break the application. Use ddtrace-run to fix this, or "
+            "import ddtrace.auto before calling gevent.monkey.patch_all().",
+            RuntimeWarning,
+        )
+
+
 EXTRA_PATCHED_MODULES = {
```

Comment:
This might be out of the scope of this PR but do we need `EXTRA_PATCHED_MODULES`? We define a list of integrations to patch here and in `ddtrace._monkey.py`. I am not sure if we need both

Reply:
Probably not. We can remove it in a separate pull request.